flexmeasures/data/models/generic_assets.py

```diff
@@ -373,6 +375,8 @@ def search_annotations(
                 annotations_before=annotations_before,
                 sources=parsed_sources,
                 annotation_type=annotation_type,
+                sensor_id=sensor_id,
```

Comment:
This variable looks odd here. See my other comment about refactoring.

Reply:
refactored.